tools/linter/adapters/_linter.py

```diff
@@ -400,21 +425,34 @@ def _replace(self, pf: PythonFile) -> tuple[str, list[LintResult]]:
 
         while True:
             try:
-                results = list(self._lint(pf))
+                results = sorted(self._lint(pf), key=LintResult.sort_key)
             except IndentationError as e:
                 error, (_name, lineno, column, _line) = e.args
-                results = [LintResult(error, lineno, column)]
-
-            if first_results is None:
-                first_results = sorted(results, key=LintResult.sort_key)
 
+                results = [LintResult(error, lineno, column)]
+                self._error(pf, *results)
+
+            except ParseError as e:
+                results = [LintResult(str(e), *e.token.start)]
+                self._error(pf, *results)
+
+            for i, ri in enumerate(results):
+                if not ri.is_recursive:
+                    for j in range(i + 1, len(results)):
+                        rj = results[j]
```

Comment:
```suggestion
                    for rj in results[i+1:]:
```
NIT, but this would be a bit cleaner the index j is only used for the lookup to assign `rj`.